x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/inference/deployment/MlPlatformArchitecturesUtilTests.java

```diff
@@ -0,0 +1,248 @@
+        var architectures = nArchitectures(randomIntBetween(2, 10));
+        String architecturesStr = architectures.toString();
+        architecturesStr = architecturesStr.substring(1, architecturesStr.length() - 1); // Remove the brackets
+        var modelId = randomAlphaOfLength(10);
+        var requiredArch = randomAlphaOfLength(10);
+        String message = "ML nodes in this cluster have multiple platform architectures, "
+            + "but can only have one for this model (["
+            + modelId
+            + "]); "
+            + "expected ["
+            + requiredArch
+            + "]; but was ["
+            + architecturesStr
+            + "]";
+
+        Throwable exception = expectThrows(
+            IllegalStateException.class,
+            "Expected IllegalStateException but no exception was thrown",
+            () -> MlPlatformArchitecturesUtil.verifyMlNodesAndModelArchitectures(architectures, requiredArch, modelId)
+        );
+        assertEquals(exception.getMessage(), message);
+    }
+
+    public void testVerifyArchitectureMatchesModelPlatformArchitecture_GivenRequiredArchMatches_ThenNothing() {
+        var requiredArch = randomAlphaOfLength(10);
+        String architecturesStr = requiredArch;
+
+        var modelId = randomAlphaOfLength(10);
+        String message = "The model being deployed (["
+            + modelId
+            + "]) is platform specific and incompatible with ML nodes in the cluster; "
+            + "expected ["
+            + requiredArch
+            + "]; but was ["
+            + architecturesStr
+            + "]";
+
+        MlPlatformArchitecturesUtil.verifyMlNodesAndModelArchitectures(
+            new HashSet<>(Collections.singleton(architecturesStr)),
+            requiredArch,
+            modelId
+        );
+    }
+
+    public void testVerifyArchitectureMatchesModelPlatformArchitecture_GivenRequiredArchDoesNotMatch_ThenThrowsIAE() {
+        var requiredArch = randomAlphaOfLength(10);
+        String architecturesStr = requiredArch + "-DIFFERENT";
+
+        var modelId = randomAlphaOfLength(10);
+        String message = "The model being deployed (["
+            + modelId
+            + "]) is platform specific and incompatible with ML nodes in the cluster; "
+            + "expected ["
+            + requiredArch
+            + "]; but was ["
+            + architecturesStr
+            + "]";
+
+        Throwable exception = expectThrows(
+            IllegalArgumentException.class,
+            "Expected IllegalArgumentException but no exception was thrown",
+            () -> MlPlatformArchitecturesUtil.verifyMlNodesAndModelArchitectures(
+                new HashSet<>(Collections.singleton(architecturesStr)),
+                requiredArch,
+                modelId
+            )
+        );
+        assertEquals(exception.getMessage(), message);
+    }
+
+    public void testVerifyArchitectureMatchesModelPlatformArchitecture_GivenTooManyArches() {
```

Comment:
TODO: update name with then part

Reply:
This test looks identical to `testVerifyMlNodesAndModelArchitectures_GivenAtLeastTwoArches_ThenThrowsISE`.

I guess one of them is supposed to test the case of multiple architectures when the required one matches one of the actual ones.